### DISCUSSION TITLE
Ping False Negatives

### DISCUSSION THREAD

**Discussion by jpotosme-sjrtd** [vercel/community#1953] [Help]
### Summary

Hello,
I have a nextjs app that was built to monitor services . It pings a list of local server's and displays the status.  I have deployed it on vercel and get false on my /api/ping response that were true locally. I am just wondering if vercel allows for such functionality? Are there any different network configuration settings that may be causing conflicts? Are there any other approaches worth pursuing?

Thank you

### Additional information

using npm node-ping package
installed with npm i ping

**Comment by amyegan** [comment]
Hi @jpotosme-sjrtd! Can you share a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) so I can get a better understanding of what is going wrong?

**Reply by jpotosme-sjrtd** [reply]
Hello @amyegan  ,
Yes here is a simplified version of what I am attempting to accomplish with no personal info.


`
  ```
import { exec } from "child_process";
  
  function runCommand(command) {
    return new Promise((resolve, reject) => {
      exec(command, (error, stdout, stderr) => {
        if (error) {
          reject(error);
        } else if (stderr) {
          reject(new Error(stderr));
        } else {
          resolve(stdout);
        }
      });
    });
  }

export default async function router(req, res) {
    if (req.method === "POST") {
      const externalIp = req.body.ip;
      
      const pingCommand = `ping ${externalIp}`;
    
      const pingResult = await runCommand(pingCommand);
      
      res.status(200).json({ ping: pingResult });
    }
  }
```
`
Results in the following error found in vercel logs:
 /bin/sh: ping: command not found at ChildProcess.exithandler . 

Is it possible to use child process from next js api endpoint on vercel?

**Reply by smaeda-ks** [reply]
@jpotosme-sjrtd Our serverless functions are backed by AWS Lambda, and they don't support ICMP outbound connections, so you can't do that:
https://repost.aws/questions/QUuZk4GSIdQcqvKrsiay82Hw/ping-url-or-dns-name-from-lambda-function

Alternatively, you could monitor via HTTP or other TCP ports. E.g., https://github.com/jethrocarr/lambda-ping

**Reply by jpotosme-sjrtd** [reply]
Thank you for the reply @smaeda-ks . Yes I discovered that the issue had to be related to the limitations set by serverless functions.
